src/lib/integration_test_lib/wait_condition.mli

```diff
@@ -31,6 +31,7 @@ module Make
     | Block_height_growth
     | Zkapp_to_be_included_in_frontier
     | Persisted_frontier_loaded
+    | Node_to_bootstrap
```

Comment:
can you just add a comment explaining what "bootstrap" means in this context, and also explain how this is different from "Node_to_initialize".

Reply:
Good point. It hard to differentiate currently as from technical side those events are basically for node initialization but are interested in different details of initialization. We have Persistent_frontier_loaded event which is a lowest level of node startup (in all initialization related events) then on slightly upper level we have 'Node_to_bootstrap' which is interested in the same event as `Persistent_frontier_loaded` but additionally want to learn about couple more information (if node dropped persistent frontier etc). Finally we have Node_to_initialize which doesn't care about those details and represent (i think) event which tells us about general node readiness.

IMHO, maybe if we rename `Node_to_bootstrap` which is too general to more specific like: `Transition_frontier_loaded` which is right name for current level of events, it will correctly inform anyone on which level of node bootstrap we are operating and what to expect. it will guide anyone that it is some more detailed than `Node_to_initialize` but no so much as  `Persisted_frontier_loaded`. what you think?

Reply:
On the whole, I agree that renaming it `Transition_frontier_loaded` and `Persisted_frontier_loaded` is better, since it is more detailed, however there are some confusing wrinkles here.

the wait condition `Nodes_to_initialize` relies on the lucy event `Node_initialization`, which looks at the daemon structured event called `Starting_transition_frontier_controller`.  This also has to do with the transition frontier being loaded, although I assume this is a separate structured event?

I'm a little unclear as to how those structured events are different, although this is a daemon things and not a lucy thing per se.  Do you have any further information on this?

Reply:
Pasting @nholland94 response for future reference:
```
From how this code is written, Transition_frontier_loaded will only get triggered if the transition frontier was successfully loaded from persistence. I believe this will happen even before bootstrap, and then the bootstrapping process will determine if that loaded frontier is too out of date (in which case we would actually run bootstrap). The Starting_transition_frontier_controller will fire whenever we finish bootstrapping and begin participating in the network, regardless of whether we used persisted frontier or executed the full bootstrap routine.
```